src/components/guide/index.js

```diff
@@ -0,0 +1,221 @@
+  }
+  introduction() {
+    const {introduction} = this.state.displayedCompetency;
+
+    const intro = introduction.split("\n", 1)[0];
+    const readMore = introduction.replace(intro, "").trim();
+    return {intro, readMore};
+  }
+  selectBoxOrTabs() {
+    const {displayedCompetency} = this.state;
+
+    if(smallScreen()) {
+      return (
+        <div>
+          <select
+            value={displayedCompetency.name}
+            className={style.mobileSelect}
+            onChange={(e) => this.displayCompetency(e)}
+          >
+            {this.state.competencies.map((competency) => (
+              <option value={competency.name}>{competency.name}</option>
+            ))
+            }
+          </select>
+          <p className={style.mobileBadge}>
+            <img src={this.tabBadge(displayedCompetency.name)} alt={`${displayedCompetency.name} badge`} />
+          </p>
+        </div>
+      );
+    } else {
+      return (
+        <ul className={style.tabs}>
+          {this.state.competencies.map((competency, index) => (
+            <li
+              className={competency.name === displayedCompetency.name ? style.tabActive : null}
+              key={competency.id}
+            >
+              <a
+                href={`#tab-${index}`}
+                tabIndex={0}
+                onKeyPress={() => this.displayCompetency(competency.name)}
+                onClick={() => this.displayCompetency(competency.name)}
+                name={competency.name}
+              >
+                <img src={this.tabBadge(competency.name)} alt={`${competency.name} badge`} />
+                <br />
+                {competency.name}
+              </a>
+            </li>
+          ))}
+        </ul>
+      );
+    }
+  }
+  setGuide() {
+    const params = {assessmentId: this.props.assessmentID, localeKey: this.props.locale};
+    this.props.traitify.graphqlQuery("/interview_guides/graphql", guideQuery({params}))
+      .then((response) => {
+        if(response.errors) {
+          this.processError(response.errors);
+          this.setState({errors: response.errors});
+          return;
+        }
+        const {competencies} = response.data.guide;
+        const badges = this.badges(competencies);
+        this.setState({badges, competencies, displayedCompetency: competencies[0]});
+      });
+  }
+  badges(competencies) {
+    const {assessment} = this.props;
+
+    return competencies.map((competency) => {
+      const personalityId = competency.questionSequences[0].personality_type_id;
+      const types = assessment.personality_types;
+      const personality = types.filter((type) => type.personality_type.id === personalityId)[0];
+      return {image: personality.personality_type.badge.image_medium, name: competency.name};
+    });
+  }
+  stringToListItems(entity) {
+    let entities = entity.split("\n");
+    entities = entities.map((e) => <li key={listId()}>{e}</li>);
```

Comment:
I'm not sure this is a good idea. I think in this case we can just disable the rule since there isn't any identifying info. You can disable it with `// eslint-disable-line whatever-rule-name`

Reply:
The key warning isnt from eslint, it's from react and only occurs in dev, I believe. I added indexes as keys and ignored that warning, but I think we should just let than warning live in dev and instead I remove what I added here. Wanted to get your thoughts first before I changed it again.